src/formats/bmp.c

```diff
@@ -383,11 +400,27 @@ enum loader_status decode_bmp(struct image* ctx, const uint8_t* data,
     color_data_sz = hdr->offset - sizeof(struct bmp_file) - bmp->dib_size;
     palette.table = color_data;
     palette.size = color_data_sz / sizeof(uint32_t);
-    mask = (color_data_sz <= sizeof(*mask) ? color_data : NULL);
+
+    // create mask
+    if (bmp->dib_size > BITMAPINFOHEADER_SIZE) {
+        mask_location = (const uint32_t*)(bmp + 1);
+
+	  mask.red = mask_location[0];
+	  mask.green = mask_location[1];
+	  mask.blue = mask_location[2];
+
+        if (bmp->dib_size > BITMAPINFOV2HEADER_SIZE) {
+            mask.alpha = mask_location[3];
+        } else {
+            mask.alpha = 0;
+        }
+    } else {
```

Comment:
I would swap if/else blocks, this will make them more readable: short one-lined `if` and then long `else`.

Reply:
I don't mind trying to fix those problems. I tested that test suite (and others, namely [this one](https://entropymine.com/jason/bmpsuite/bmpsuite/html/bmpsuite.html)) before, and I didn't realize that particular image was broken, my bad.

I must note that many other images are (and were already) broken, namely RLE images with delta codes. I am someone who loves reinventing the wheel and writing my own implementations for things that already have one. However, even if the BMP format seems simple, it certainly isn't as simple as people think. Have you considered using a library for perfect compatibility?

Reply:
> I am someone who loves reinventing the wheel and writing my own implementations
Yeah, me too =)

> even if the BMP format seems simple, it certainly isn't as simple as people think
Exactly, I've spent a lot of time writing the current implementation and it still doesn't cover all things.

> Have you considered using a library for perfect compatibility?
This should be a library that is available on most Linux distributions and has a C interface.
I didn't find anything suitable. That's why I have my own implementation.